pytype/abstract/class_mixin.py

```diff
@@ -552,20 +568,31 @@ def _get_attrs_from_mro(self, cls_attrs, metadata_key):
 
     all_attrs = {}
     sub_attrs = []
+    type_params = {}
     attributes_to_ignore = set()
     for base_cls in reversed(self.mro[1:]):
       if not isinstance(base_cls, Class):
         continue
       # Some third-party dataclass implementations add implicit fields that
       # should not be considered inherited attributes.
       attributes_to_ignore.update(getattr(base_cls, "IMPLICIT_FIELDS", ()))
+      # Any subclass of a Parameterized dataclass must inherit attributes from
+      # its parent's init.
+      # See https://github.com/google/pytype/issues/1104
+      if isinstance(base_cls, _classes.ParameterizedClass):
+        type_params = base_cls._formal_type_parameters
```

Comment:
use `base_cls.formal_type_parameters`, which takes care of lazily loading the attribute.